packages/toml/src/index.ts

```diff
@@ -0,0 +1,82 @@
+import { Taplo } from '@taplo/lib'
+import type { Plugin } from 'prettier'
+
+import { defaultOptions, options } from './options'
+import type { AST, PrettierOptions, TaploOptions } from './types'
+
+const LANGUAGE = 'toml'
+const PARSER = 'toml'
+const AST_NAME = 'toml-ast'
+
+let taplo: Taplo | null = null
+
+async function format(code: string, options: TaploOptions) {
+  if (!taplo) {
+    taplo = await Taplo.initialize()
+  }
+
+  return taplo.format(code, { options })
+}
+
+function removeBeginningTrailingNewline(code: string) {
+  code = code.replace(/^(?:\r?\n)+/, '')
+  code = code.replace(/(?:\r?\n)+$/, '')
+
+  return code
+}
+
+const TomlPlugin: Plugin<AST> = {
+  languages: [
+    {
+      name: LANGUAGE,
+      parsers: [PARSER],
+    },
+  ],
+  parsers: {
+    [PARSER]: {
+      async parse(code: string, options: PrettierOptions) {
+        const indentString = options.useTabs
+          ? '\t'
+          : ' '.repeat(options.tabWidth)
+
+        const taploOptions: TaploOptions = {
```

Comment:
Can we remove all `taplo` prefix in favour of prettier's `overrides` feature? And just use deconstructing for default options values.